cable/courier/anycable.go

```diff
@@ -33,50 +28,72 @@ type EventData struct {
 }
 
 type Client struct {
-	Redis *redis.Client
+	Redis        *redis.Client
+	RedisChannel string
 }
 
-func NewClient(rdc *redis.Client) *Client {
-	return &Client{Redis: rdc}
+func NewClient(rdc *redis.Client, redisChannel string) *Client {
+	return &Client{Redis: rdc, RedisChannel: redisChannel}
 }
 
-func (c *Client) BroadcastMessage(msgName string, msg proto.Message, stream, correlationID string) {
-	msgJSON := newEventJSONFromMessage(msgName, msg, stream, correlationID)
+func (c *Client) BroadcastMessage(msgName string, msg proto.Message, stream, correlationID string) error {
+	msgJSON, err := newEventJSONFromMessage(msgName, msg, stream, correlationID)
+	if err != nil {
+		return fmt.Errorf("failed to marshal anycable message: %w", err)
+	}
+
 	c.publish(msgJSON)
+
+	return nil
 }
 
 func (c *Client) publish(msg string) {
-	c.Redis.Publish(context.Background(), RedisChannel, msg)
+	c.Redis.Publish(context.Background(), c.RedisChannel, msg)
 }
 
-// I'm really sorry about loads of `_` in this function. Proper error handling seemed too much for this task at the moment.
-// Nevertheless, it should just work if not touched, trust me :)
-//
-// TODO: Add proper error handling
-func newEventJSONFromMessage(msgName string, msg protoreflect.ProtoMessage, stream string, correlationID string) string {
-	res, _ := protojson.Marshal(msg)
+func newEventJSONFromMessage(msgName string, msg protoreflect.ProtoMessage, stream string, correlationID string) (string, error) {
+	res, err := protojson.Marshal(msg)
+	if err != nil {
+		return "", err
```

Comment:
It would be great to add more context for errors in this function, e.g.:

```suggestion
		return "", fmt.Errorf("failed to marshal proto message: %w", err)
```

Reply:
Fixed